src/classes/Cluster.ts

```diff
@@ -0,0 +1,352 @@
+import { Nullable } from "../types";
+import { MarkerClustering } from "./MarkerClustering";
+
+/**
+ * 마커를 가지고 있는 클러스터를 정의합니다.
+ */
+export class Cluster {
+  _clusterCenter: Nullable<naver.maps.Coord> = null;
+  _clusterBounds: Nullable<naver.maps.LatLngBounds> = null;
+  _clusterMarker: Nullable<naver.maps.Marker> = null;
+  _relation: Nullable<naver.maps.MapEventListener> = null;
+  _clusterMember: Array<naver.maps.Marker> = [];
+  _markerClusterer: Nullable<MarkerClustering> = null;
+
+  constructor(markerClusterer: MarkerClustering) {
+    this._clusterCenter = null;
+    this._clusterBounds = null;
+    this._clusterMarker = null;
+    this._relation = null;
+
+    this._clusterMember = [];
+
+    this._markerClusterer = markerClusterer;
+  }
+
+  /**
+   * 클러스터에 마커를 추가합니다.
+   * @param marker 클러스터에 추가할 마커
+   */
+  addMarker(marker: naver.maps.Marker) {
+    if (this._isMember(marker)) return;
+
+    if (!this._clusterCenter) {
+      const position = marker.getPosition();
+      this._clusterCenter = position;
+
+      const bounds = this._calcBounds(position);
+      if (bounds) this._clusterBounds = bounds;
+    }
+
+    this._clusterMember.push(marker);
+  }
+
+  /**
+   * 클러스터를 제거합니다.
+   */
+  destroy() {
+    if (this._relation) naver.maps.Event.removeListener(this._relation);
+
+    const members = this._clusterMember;
+
+    for (let i = 0, ii = members.length; i < ii; i++) {
```

Comment:
```suggestion
    for (let i = 0, i < members.length; i++) {
```

따로 변수를 사용하는 곳이 없는 것 같은데 이렇게 하는건 어떤가요?